edumanage/forms.py

```diff
@@ -195,6 +197,38 @@ def clean_host(self):
             raise forms.ValidationError(_('This field is required.'))
         return self.cleaned_data["host"]
 
+    def clean_psk_identity(self):
+        psk_identity = self.cleaned_data['psk_identity']
+        if not 'proto' in self.cleaned_data:
+                raise forms.ValidationError(_('The Protocol field is required to validate this field.'))
+        proto = self.cleaned_data['proto']
+        if not 'ertype' in self.cleaned_data:
+                raise forms.ValidationError(_('The Type field is required to validate this field.'))
+        ertype = self.cleaned_data['ertype']
+        if proto in RADPROTOS.TLSPSK:
+            if ertype in ERTYPE_ROLES.IDP:
+                if psk_identity:
+                    match = re.match(NAI_RE, psk_identity)
+                    if not match:
+                        raise forms.ValidationError(_('Invalid Network Access Identifier format.'))
+                    return self.cleaned_data["psk_identity"]
+                else:
+                    raise forms.ValidationError(_('This field is required for IdPs.'))
+
+    def clean_psk_key(self):
+        psk_key = self.cleaned_data['psk_key']
+        if not 'proto' in self.cleaned_data:
+                raise forms.ValidationError(_('The Protocol field is required to validate this field.'))
+        proto = self.cleaned_data['proto']
+        if proto in RADPROTOS.TLSPSK:
+            if psk_key:
+                if len(psk_key) > 16:
```

Comment:
This would reject a key of exactly 16 octets - how about:
```suggestion
                if len(psk_key) >= 16:
```

Reply:
Doh, that's a stupid oversight. Comes from never using < 20 chars to test :)